### ISSUE TITLE
Softmax activation layer

### ISSUE THREAD

**Issue by dmccloskey** [dmccloskey/EvoNet#55]
### Objectives
- [x] Inverse activation
- [x] Exponential activation
- [x] Softmax module

### References
- https://deepnotes.io/softmax-crossentropy
- https://stats.stackexchange.com/questions/265905/derivative-of-softmax-with-respect-to-weights
